### DISCUSSION TITLE
Service provider configuration

### DISCUSSION THREAD

**Discussion by sparkoo** [redhat-appstudio/service-provider-integration-operator#427] [General]
There are 3 levels how we can configure service providers. In priority order (1 is highest):

1. labeled Secret in user's namespace
2. config file set with `--config-file` cli arg
3. defaults hardcoded in codebase

Each service provider type can have multiple configurations on the same level, but they must have different host. When host is not explicitly set, we assume default SAAS host (`github.com`, `quay.io`, `gitlab.com`).

Service providers can have some capabilities:
 - OAuth (defined by configuration)
   - only in cases when we have OAuth configuration, OAuth URL is generated into `SPIAccessToken` (and derived objects) status
 - Download File (defined in code)
 - Refresh Token (defined in code)

**Comment by gazarenkov** [comment]
Nice, it is useful documentation, I'd just make it more clear that Configuration and Capabilities are almost unbound to each other (except the OAuth aspect)
